Document ThrowOnEachFailureUnderDebugger option (#947)
nunit/docs: docs/articles/nunit/technical-notes/usage/Debugging-Support.md

# Debugging Support

This document describes framework functionality for debugging support of tests and tested code.

## `ThrowOnEachFailureUnderDebugger` Setting

(From version 4.2.0)

Setting name defined by `FrameworkPackageSettings.ThrowOnEachFailureUnderDebugger` constant.

> This boolean framework setting enables throwing of immediately catched assertion exception from `Assert.Multiple`

Reviewer: ```suggestion
This boolean framework setting enables throwing exceptions in `Assert.Multiple` immediately upon being caught.
```

> This boolean framework setting enables throwing of immediately catched assertion exception from `Assert.Multiple`
> block on each failed assertion. This allows user to break execution on exception and inspect tested code state
> at time of failure. In contrast default behavior of `Assert.Multiple` is to throw assertion exception only after
> whole block executed. In this situation tested code state could change or be unavailable.

Reviewer: ```suggestion

It may sometimes be necessary to debug the current state of a test during an `Assertion.Multiple` block. Typically, assertion exceptions are thrown only after the block is completed.

This boolean framework setting will cause the debugger to throw on a failed assertion during an `Assert.Multiple`
block. This allows user to break execution on exception and inspect tested code state
at time of failure.
```

# Debugging Support

This document describes framework functionality for debugging support of tests and tested code.

## `ThrowOnEachFailureUnderDebugger` Setting

(From version 4.2.0)

Setting name defined by `FrameworkPackageSettings.ThrowOnEachFailureUnderDebugger` constant.

> 
> 

Reviewer: ```suggestion

```

Removing extra blank line.
